Library/ViewModels/PledgeViewModel.swift

```diff
@@ -338,6 +361,70 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
     let createBackingTransactionSuccess = project.takeWhen(createBackingEventSuccess)
 
     self.goToThanks = Signal.merge(applePayTransactionCompleted, createBackingTransactionSuccess)
+
+    let amountChanged = Signal.combineLatest(
+      project.map { $0.personalization.backing?.pledgeAmount },
+      self.pledgeAmountDataSignal.map { $0.0 }
+    )
+    .map(!=)
+
+    let shippingRuleChanged = Signal.merge(
+      self.viewDidLoadProperty.signal.mapConst(false),
+      Signal.combineLatest(
+        project.map { $0.personalization.backing?.locationId },
+        self.shippingRuleSelectedSignal.map { $0.location.id }
+      )
+      .map(!=)
+    )
+
+    let valuesChanged = Signal.combineLatest(
+      amountChanged,
+      shippingRuleChanged,
+      self.viewDidLoadProperty.signal
```

Comment:
I _think_ you don't need to combine with `viewDidLoad` because `amountChanged` and `shippingRuleChanged` both already depend on `viewDidLoad` 🤔. You could easily test this by removing `viewDidLoad` from here - your tests should still pass.